array.c

```diff
@@ -5920,23 +5920,24 @@ ary_recycle_hash(VALUE hash)
 
 /*
  *  call-seq:
- *     ary - other_ary    -> new_ary
+ *    array - other_array -> new_array
  *
- *  Array Difference
- *
- *  Returns a new array that is a copy of the original array, removing all
- *  occurrences of any item that also appear in +other_ary+. The order is
- *  preserved from the original array.
- *
- *  It compares elements using their #hash and #eql? methods for efficiency.
+ *  Argument +other_array+ must be an
+ *  {Array-convertible object}[doc/implicit_conversion_rdoc.html#label-Array-Convertible+Objects].
```

Comment:
I recommend we move the link below where it discusses the exception raised.  It's better to have a description of what the method does first.  The same goes for other modified methods.

Reply:
There are many dozens of these, not just in Array, but also merged into Hash, ENV, JSON, and CSV.  So changing them would be a task.

Reply:
An ideal schema might be:
- call-seq.
- One-line description, no details, no code.
- Requirements on arguments.
- Details with examples.
- Exceptions with examples.

That would require a new pass over everything, but we could consider it.  My aim is to enhance.

Reply:
It just seems odd to have the first thing in the documentation address a error case.  The call-seq implies the type of the other argument already, and any-non array convertible object is going to be an error.

In general, my ideal schema would be:

* call-seq
* description
* details/examples
* detailed argument description (for complex cases, probably omitted for simple cases)
* exceptions (only if non-obvious)

However, it would probably be a good idea to get feedback from other committers.

Reply:
I know more now about this than I did when I started (HL: human learning).  I've thought more than once about proposing that we create a formal style guide (for ruby/ruby, not for  Ruby).  Sometimes those reviewing what I write have various opinions.

Reply:
For now, I'd prefer to leave the arg remark as is, to be consistent with many others.

Reply:
That's fine.  We can always do a more mechanical style cleanup later.